Applications/Utils/MeshEdit/flipNodeCoordinates.cpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * @copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/LICENSE.txt
+ *
+ */
+
+#include <memory>
+#include <string>
+#include <valarray>
+
+#include <tclap/CmdLine.h>
+
+#include "Applications/ApplicationsLib/LogogSetup.h"
+
+#include "BaseLib/BuildInfo.h"
+
+#include "MeshLib/Node.h"
+#include "MeshLib/Mesh.h"
+
+#include "MeshLib/IO/readMeshFromFile.h"
+#include "MeshLib/IO/writeMeshToFile.h"
+
+void flipNodeCoordinates(MeshLib::Mesh &mesh, std::array<int, 3> const& new_coord_ids)
+{
+    double new_xyz[3] = {};
+    for (MeshLib::Node* node : mesh.getNodes())
+    {
+        for (int i=0; i<3; i++)
+            new_xyz[i] = (*node)[new_coord_ids[i]];
+        node->updateCoordinates(new_xyz[0], new_xyz[1], new_xyz[2]);
```

Comment:
you are right, exterior functions shouldn't be declared as friends. I'll change it to use `Mesh::map_nodes(std::function&&)`.